From 952d067ae1367cf04ae5733446aeec19ba6127fd Mon Sep 17 00:00:00 2001
From: Henrique Ferreiro <hferreiro@igalia.com>
Date: Fri, 12 Jun 2020 19:54:21 +0000
Subject: Add missing cstring include and namespace in WebString

WebString was using strlen without the proper includes. Apparently, this
broke some external code.

Change-Id: Ib01efa6bf3fe72e4f27e2f43641d0157dd457b9a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2243133
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Commit-Queue: Henrique Ferreiro <hferreiro@igalia.com>
Cr-Commit-Position: refs/heads/master@{#777929}
---
 third_party/blink/public/platform/web_string.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/third_party/blink/public/platform/web_string.h b/third_party/blink/public/platform/web_string.h
index a74c413d5982..b9f774c190c5 100644
--- a/third_party/blink/public/platform/web_string.h
+++ b/third_party/blink/public/platform/web_string.h
@@ -31,7 +31,9 @@
 #ifndef THIRD_PARTY_BLINK_PUBLIC_PLATFORM_WEB_STRING_H_
 #define THIRD_PARTY_BLINK_PUBLIC_PLATFORM_WEB_STRING_H_
 
+#include <cstring>
 #include <string>
+
 #include "base/memory/scoped_refptr.h"
 #include "base/optional.h"
 #include "base/strings/latin1_string_conversions.h"
@@ -109,7 +111,7 @@ class WebString {
   BLINK_PLATFORM_EXPORT bool Equals(const WebString&) const;
   BLINK_PLATFORM_EXPORT bool Equals(const char* characters, size_t len) const;
   bool Equals(const char* characters) const {
-    return Equals(characters, characters ? strlen(characters) : 0);
+    return Equals(characters, characters ? std::strlen(characters) : 0);
   }
 
   BLINK_PLATFORM_EXPORT size_t length() const;
-- 
2.20.1

